templates/community-viz/scripts/message.js

```diff
@@ -0,0 +1,85 @@
+// imports
+const path = require('path');
+const webpack = require('webpack');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const program = require('commander');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'));
+
+// constants
+const DEV_BUCKET = process.env.npm_package_config_gcsDevBucket;
+const PROD_BUCKET = process.env.npm_package_config_gcsProdBucket;
+const MANIFEST_FILE = process.env.npm_package_config_manifestFile;
+const CSS_FILE = process.env.npm_package_config_cssFile;
+const JSON_FILE = process.env.npm_package_config_jsonFile;
+
+// configure command-line arguments
+program
+  .option('-p, --prod', 'Build for prod?', /^(true|false)$/i, true)
+  .parse(process.argv);
+
+// default to dev if it's not prod
+const DEVMODE = program.args[0] === 'true' ? false : true;
+const GCS_BUCKET = DEVMODE ? DEV_BUCKET : PROD_BUCKET;
+
+const encoding = 'utf-8';
+
+// common options
+let webpackOptions = {
+  entry: {
+    // this is the viz source code
+    main: path.resolve(__dirname, '..', 'src', 'printMessage.js'),
+  },
+  output: {
+    filename: 'index.js',
+    path: path.resolve(__dirname, '..', 'datastudio'),
+  },
+  plugins: [
+    new CopyWebpackPlugin([
+      {from: path.join('src', JSON_FILE), to: '.'},
+      {from: path.join('src', CSS_FILE), to: '.'},
+    ]),
+  ],
+};
+
+if (DEVMODE === true) {
+  const devOptions = {
+    mode: 'development',
+    devtool: 'inline-source-map',
+  };
+  webpackOptions = Object.assign(webpackOptions, devOptions);
```

Comment:
jsyk: Object.assign will mutate the first object you give it. So for this you could just do `Object.assign(webpackOptions, devOptions)`.